pkg/network/placement_configuration_test.go

```diff
@@ -0,0 +1,95 @@
+)
+
+var _ = Describe("Testing PlacementConfiguration", func() {
+	var (
+		defaultPlacementConfiguration = cnao.PlacementConfiguration{
+			Infra: &cnao.Placement{
+				NodeSelector: map[string]string{
+					"beta.kubernetes.io/arch":        "amd64",
+					"node-role.kubernetes.io/master": "",
+				},
+				Tolerations: []corev1.Toleration{
+					corev1.Toleration{
+						Key:      "node-role.kubernetes.io/master",
+						Operator: corev1.TolerationOpExists,
+						Effect:   corev1.TaintEffectNoSchedule,
+					},
+				},
+			},
+			Workloads: &cnao.Placement{
+				NodeSelector: map[string]string{
+					"beta.kubernetes.io/arch": "amd64",
+				},
+				Tolerations: []corev1.Toleration{
+					corev1.Toleration{
+						Operator: corev1.TolerationOpExists,
+						Effect:   corev1.TaintEffectNoSchedule,
+					},
+				},
+			},
+		}
+	)
+	type fillDefaultsCase struct {
+		previousConfig *cnao.NetworkAddonsConfigSpec
+		currentConfig  *cnao.NetworkAddonsConfigSpec
+		expectedConfig cnao.PlacementConfiguration
+	}
+	DescribeTable("fill defaults function",
+		func(c fillDefaultsCase) {
+			errorList := fillDefaultsPlacementConfiguration(c.currentConfig, c.previousConfig)
+			Expect(*c.currentConfig.PlacementConfiguration).To(Equal(c.expectedConfig))
+			Expect(errorList).To(BeEmpty())
+		},
+		Entry("when PlacementConfiguration is nil should return default values", fillDefaultsCase{
+			previousConfig: &cnao.NetworkAddonsConfigSpec{},
+			currentConfig:  &cnao.NetworkAddonsConfigSpec{},
+			expectedConfig: defaultPlacementConfiguration,
+		}),
+		Entry("when Workloads PlacementConfiguration is nil should return default values", fillDefaultsCase{
+			previousConfig: &cnao.NetworkAddonsConfigSpec{},
+			currentConfig: &cnao.NetworkAddonsConfigSpec{
+				PlacementConfiguration: &cnao.PlacementConfiguration{
+					Infra: &cnao.Placement{},
+				},
+			},
+			expectedConfig: defaultPlacementConfiguration,
+		}),
+		Entry("when Infra PlacementConfiguration is nil should return default values", fillDefaultsCase{
+			previousConfig: &cnao.NetworkAddonsConfigSpec{},
+			currentConfig: &cnao.NetworkAddonsConfigSpec{
+				PlacementConfiguration: &cnao.PlacementConfiguration{
+					Workloads: &cnao.Placement{},
+				},
+			},
+			expectedConfig: defaultPlacementConfiguration,
+		}),
+		Entry("when the user hasn't explicitly changed PlacementConfiguration and a previous configuration exits should use the previous values on the current one", fillDefaultsCase{
+			previousConfig: &cnao.NetworkAddonsConfigSpec{
+				PlacementConfiguration: &cnao.PlacementConfiguration{
+					Workloads: &cnao.Placement{
+						NodeSelector: map[string]string{
+							"beta.kubernetes.io/arch": "amd64",
+						},
+					},
+				},
+			},
+			currentConfig: &cnao.NetworkAddonsConfigSpec{},
+			expectedConfig: cnao.PlacementConfiguration{
+				Workloads: &cnao.Placement{
+					NodeSelector: map[string]string{
+						"beta.kubernetes.io/arch": "amd64",
+					},
+				},
+			},
+		}),
```

Comment:
I miss a test checking that we reject changes in the configuration

Reply:
Right, added.